src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -1498,11 +1571,56 @@ public void Dispose()
             // Do this here because otherwise if a purge job runs there will
             // be a deadlock while the purge job waits for the lock that is held
             // by the disposal which is waiting for timer callbacks to finish
-            StopExpirePurgeTimer(); 
-            ThreadSafety.DoLocked(() =>
-            {
-                ThrowIfActiveItems();
-                Dispose(true);
+            StopExpirePurgeTimer();
+            ThreadSafety.DoLocked(() => {
+                if (IsClosed) {
+                    return;
+                }
+
+                if (!_isClosing) {
+                    _isClosing = true;
+                }
+            });
+
+            _closeCondition.Reset();
+            foreach (var q in ActiveLiveQueries) {
+                q.Key.Stop();
+            }
+
+            while (!IsReadyToClose) {
+                _closeCondition.WaitOne();
```

Comment:
This loop is not valid.  It will wait once, and then have a tight loop because the condition remains set forever.  Consider using `AutoResetEvent` instead so that each time you wait, it will automatically reset.  Furthermore, it would be good to dispose it in the database dispose method.